### ISSUE TITLE
Get added to 'apt' and 'yum' repositories

### ISSUE THREAD

**Issue by kiloforce** [wavetermdev/waveterm#80]
'apt' and 'yum' are the most common ways to install packages.  Should look at getting Wave added to those repositories for like Ubuntu and Fedora.

**Comment by sawka** [comment]
Yes, this is on the roadmap.  Needed to get some stars on the repo before they'd accept us :).

However, we'll probably start with "snap" and "flatpak" first since they are easier for binary distributions.

**Comment by gonzolively** [comment]
Related: #71 

And my vote is heavily weighted towards flatpak...most developers are pretty peeved with the way Canonical has handled the SNAP format. Plus, performance is clearly worse than flatpak, less universal, and ultimately owned (and shoved down users throats) by Canonical.

**Comment by esimkowitz** [comment]
We're not going to bother with getting into Apt, the requirements are frankly insanely hard to meet. We'd need to create an entirely different build pipeline. We're going to go for Snap and Flatpak instead

**Comment by RokeJulianLockhart** [comment]
> Yum is also probably not worth it.

@esimkowitz, why? By the way, it's very strange that you're discussing `yum` instead of `dnf`, when `dnf` replaced `yum` years ago. Perhaps any documentation you saw about `yum` has provided you with an outdated view of RPM? `dnf5` is the most recent version, in Fedora 41. It comes with significant changes.

Additionally, you needn't try to get your packages into the official repositories. I don't know whether the original issue reporter wanted that, but if they did, it's reductive. The goal is to have the `.Deb` and `.RPM` packages update automatically from a source. Flatpak and Snap trust the developer to do so, and the equivalent exists for the aforementioned native package types too:

OS | Official Repository | Counterpart for Trusted Authors
-|-|-
Fedora | https://packages.fedoraproject.org/ [^1] | https://copr.fedorainfracloud.org/ [^2]
Ubuntu | https://packages.ubuntu.com/ | https://askubuntu.com/revisions/71516/4

[^1]: https://discussion.fedoraproject.org/t/how-can-i-add-software-to-the-fedora-distribution/69007/2
[^2]: https://docs.fedoraproject.org/en-US/quick-docs/publish-rpm-on-copr/

I don't know what CI you use, but if you used something simple like the openSUSE Build Service, you could use their templates for all of your Linux packages, and provide repositories at the same time. [^3] That's the most popular solution to this problem.

[^3]: https://en.opensuse.org/openSUSE:Build_Service_Tutorial#Terminology:~:text=The%20output%20of%20a%20project%20is%20one%20or%20more%20repositories.

**Comment by esimkowitz** [comment]
I'm reopening this, we'll look into getting into the less-restrictive repos

**Comment by kiloforce** [comment]
Updated to reflect, thanks
